packages/cardano-services/src/ChainHistory/DbSyncChainHistory/ChainHistoryBuilder.ts

```diff
@@ -261,7 +261,7 @@ export class ChainHistoryBuilder {
     if (result.rows.length === 0) return [];
 
     const txOutIds = result.rows.flatMap((txOut) => BigInt(txOut.id));
-    const multiAssets = await this.queryMultiAssetsByTxOut(txOutIds);
+    const multiAssets = collateral ? new Map() : await this.queryMultiAssetsByTxOut(txOutIds);
```

Comment:
Could you please elaborate on the reason of this change? It might be an issue when inspecting transaction on front-end. We're computing the difference of `consumed` and `produced` coins and assets that come from and goes to our own addresses. This could make it look like we've spent some assets on a failed transaction.

Reply:
This can't be source of any issue: we load `collaterReturn` just for additional info, today with assets from a random output from a random transaction from a random address, after this change simply without any assets.
Transaction with failed phase 2 validation have their `collaterReturn` stored in db-sync as a standard output, so they already have the correct map of assets regardless from this fix.

Reply:
Not sure I understand. Are we still returning `collateralReturn: { value: { coins, assets } }` with assets for chain history requests?

Reply:
We are returning `collateralReturn: Cardano.TxOut` anyway, but today the asset property contains random assets after the fix it is empty.
Anyway this can't be source of any issue as `collateralReturn`
- is never used for transaction with valid contract
- is stored (in db-sync) as an output so is correctly read with the right assets

Reply:
d1ead2edd4a should solve this